homeassistant/block_async_io.py

```diff
@@ -46,53 +49,139 @@ def _check_sleep_call_allowed(mapped_args: dict[str, Any]) -> bool:
     return False
 
 
+@dataclass(slots=True, frozen=True)
+class BlockingCall:
+    """Class to hold information about a blocking call."""
+
+    original_func: Callable
+    module: ModuleType
```

Comment:
Why do we type this as `ModuleType` and then override mypy warnings when we want to patch methods?

Reply:
`HTTPConnection` is typed as `type[HTTPConnection]`

homeassistant/block_async_io.py:68: error: Argument "module" to "BlockingCall" has incompatible type "type[HTTPConnection]"; expected Module  [arg-type]

Reply:
Yes, that was my point.
Why do we type this as module when we're not only replacing attributes of modules but also of classes?

Reply:
Since it allows patching any object and not only modules, I changed it to `object`